Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
+def remove_member():
+    team_id = demisto.args().get('team_id')
+    membership_id = demisto.args().get('membership_id')
+    remove_member__request(team_id, membership_id)
+    demisto.results(f'Team member {membership_id} was removed from the team {team_id} successfully.')
+
+
+@logger
+def update_member_request(team_id: str, membership_id: str, roles: Optional[list] = None) -> dict:
+    url = f'{GRAPH_BASE_URL}/v1.0/teams/{team_id}/members/{membership_id}'
+    return cast(Dict[Any, Any], http_request(
+        'PATCH',
+        url,
+        json_={
+            '@odata.type': '#microsoft.graph.aadUserConversationMember',
+            'roles': roles
+        }
+    ))
+
+
+def update_member():
+    team_id = demisto.args().get('team_id', '')
+    membership_id = demisto.args().get('membership_id', '')
+    team_member = update_member_request(
+        team_id=team_id,
+        membership_id=membership_id,
+        roles=['owner'] if argToBoolean(demisto.args().get('is_owner', 'false')) else [],
+    )
+    return_results(CommandResults(
+        outputs_prefix='MicrosoftTeams.TeamMember',
+        outputs_key_field='id',
+        readable_output=tableToMarkdown(
+            f'Team member {membership_id} was updated successfully.',
+            team_member,
+            ['id', 'displayName', 'email', 'roles'],
+        ),
+        outputs=team_member,
+        raw_response=team_member,
+    ))
+
+
+@logger
+def archive_team_request(team_id: str) -> None:
+    url = f'{GRAPH_BASE_URL}/v1.0/teams/{team_id}/archive'
+    http_request('POST', url)
+
+
+def archive_team():
+    team_id = demisto.args().get('team_id')
+    archive_team_request(team_id)
+    demisto.results(f'Team {team_id} was archived successfully.')
+
+
+@logger
+def unarchive_team_request(team_id: str) -> None:
+    url = f'{GRAPH_BASE_URL}/v1.0/teams/{team_id}/unarchive'
+    http_request('POST', url)
+
+
+def unarchive_team():
+    team_id = demisto.args().get('team_id')
+    unarchive_team_request(team_id)
+    demisto.results(f'Team {team_id} was unarchived successfully.')
+
+
+@logger
+def clone_team_request(
+        team_id: str,
+        display_name: str,
+        description: Optional[str] = None,
+        visibility: Optional[str] = None,
+        parts_to_clone: Optional[str] = None,
+) -> None:
+    url = f'{GRAPH_BASE_URL}/v1.0/teams/{team_id}/clone'
+    http_request(
+        'POST',
+        url,
+        json_={
+            'displayName': display_name,
+            # from Microsoft docs: "mailNickname property must be specified when a group is created.
+            # If this property is not specified, it will be computed from the displayName.
+            # Known issue: this property is currently ignored."
+            'mailNickname': display_name,
+            'description': description,
+            'visibility': visibility,
+            'partsToClone': parts_to_clone,
+        }
+    )
+
+
+def clone_team():
+    team_id = demisto.args().get('team_id')
```

Comment:
don't call `demisto.args()` every time, save that value as a variable.
same for all